src/functions.php

```diff
@@ -776,7 +776,7 @@ function _parse_message($message)
 
     list($startLine, $rawHeaders) = $headerParts;
 
-    if (preg_match("/HTTP\/(\d+(?:\.\d+)?)/i", $startLine, $matches) && $matches[1] === '1.0') {
+    if (preg_match("/(?:^HTTP\/|^[A-Z]+ \S+ HTTP\/)(\d+(?:\.\d+)?)/i", $startLine, $matches) && $matches[1] === '1.0') {
```

Comment:
You can simply use the regex from `http-server`: `if (!\preg_match("/^([A-Z]+) (\S+) HTTP\/(\d+(?:\.\d+)?)$/i", $startLine, $matches)) {`

Reply:
No, I can't. This function parses both requests and responses.

Reply:
`HTTP/1.0 200` vs `GET / HTTP/1.0`